### PR TITLE
Add JSON RPC handling for listscripts wallet command

### PR DESCRIPTION
**Pull request by cjepson**
Framework handling the new listscripts JSON RPC command for wallet
has been added. This will allow wallet users to dump the complete list
of redeem scripts from their wallet.

### CODE REVIEW

File: dcrjson/dcrdextresults.go [hunk 1]
```diff
@@ -5,6 +5,18 @@
...
 
+// ScriptInfo is the structure representing a redeem script, its hash,
+// and its address.
+type ScriptInfo struct {
+	Hash160      string `json:"hash160"`
+	Address      string `json:"address"`
+	RedeemScript string `json:"redeemscript"`
+}
+
+type ListScriptsResult struct {
```
Comment (jcvernaleo) [inline dcrjson/dcrdextresults.go]: This one is exported so it needs a comment.

**Comment by cjepson** [comment]
It was in the wrong file too, moved it to wallet extended results.

**Comment by alexlyp** [comment]
utACK

**Comment by jcvernaleo** [comment]
OK

**Pull request merged**